frontend/src/scenes/insights/ActionFilter/entityFilterLogic.ts

```diff
@@ -0,0 +1,183 @@
+import { kea } from 'kea'
+import { actionsModel } from '~/models/actionsModel'
+import {
+    EntityTypes,
+    FilterType,
+    Entity,
+    EntityType,
+    ActionFilter,
+    EntityFilter,
+    Optional,
+    PropertyFilter,
+    ActionType,
+} from '~/types'
+import { entityFilterLogicType } from './entityFilterLogicType'
+import { ActionFilterProps } from './ActionFilter'
+import { eventDefinitionsLogic } from 'scenes/events/eventDefinitionsLogic'
+
+export type LocalFilter = EntityFilter & { order: number; properties?: PropertyFilter[] }
+export type BareEntity = Pick<Entity, 'id' | 'name'>
+
+type FilterWithOptionalInsight = Optional<FilterType, 'insight'>
```

Comment:
See comment on types about `FilterType`, but if you keep the type unchanged from what you have `insight` is already optional.